### PR TITLE
Validate redeem

### PR DESCRIPTION
**Pull request by fgregoribattista**
validate withdraw allowed and refact general error dialog

### CODE REVIEW

**Review by galimba — APPROVED** [review]
LGTM

File: src/components/dialog/withdraw/WithdrawInput.vue [hunk 1]
```diff
@@ -207,12 +213,30 @@ export default {
...
     withdraw() {
       this.waiting = true;
       this.$emit("wait");
-      this.data.market
-        //TODO false is tru if max press(validate)
-        .withdraw(this.amount, false)
+      //validate withdrawAllowed
+      this.withdrawAllowed()
+        .then((allowed) => {
+          console.log(allowed);
```
Comment (pmprete) [inline src/components/dialog/withdraw/WithdrawInput.vue:234]: remove console.log
```suggestion
```

File: src/components/dialog/withdraw/WithdrawInput.vue [hunk 2]
```diff
@@ -223,7 +247,11 @@ export default {
         })
         .catch((error) => {
           this.waiting = false;
-          this.$emit("error");
+          //validate user error message
+          let userError = typeof error === "string" ? error : null;
```
Comment (pmprete) [inline src/components/dialog/withdraw/WithdrawInput.vue:251]: if error is not a string use error.message || ''

File: src/components/dialog/supply/SupplyDialog.vue [hunk 3]
```diff
@@ -104,11 +112,14 @@ export default {
...
       this.hash = null;
       this.errorDialog = null;
+      this.userErrorMessage = null;
     },
-    actionError() {
+    actionError(errorObject) {
       this.succeed = false;
       this.waiting = false;
       this.errorDialog = true;
+      this.userErrorMessage = (!errorObject.userErrorMessage) ? null : errorObject.userErrorMessage;
```
Comment (pmprete) [inline src/components/dialog/supply/SupplyDialog.vue:121]: it's easier to use the `errorObject.userErrorMessage || null ` operator than ternary operation

File: src/middleware/middleware.js [hunk 4]
```diff
@@ -99,4 +100,10 @@ export default class Middleware {
     return totalsReduced;
   }
 
+  getMsjErrorCodeComptroller(errorNumber, isErroInfo = false) {
+    errorNumber = new BigNumber(errorNumber).toNumber();
```
Comment (pmprete) [inline src/middleware/middleware.js:104]: maybe it's better to just use Number(errorNumber) instead of using BigNumber and then casting it to Number

**Review by pmprete — APPROVED** [review]
Except some minor comments looks good

File: src/middleware/middleware.js (hunk 4 above)
Comment (fgregoribattista) [reply src/middleware/middleware.js]: yes it's better. Applied

File: src/components/dialog/withdraw/WithdrawInput.vue (hunk 2 above)
Comment (fgregoribattista) [reply src/components/dialog/withdraw/WithdrawInput.vue:250]: for this case, error.message is better. Applied

File: src/components/dialog/supply/SupplyDialog.vue (hunk 3 above)
Comment (fgregoribattista) [reply src/components/dialog/supply/SupplyDialog.vue]: done!

**Comment by fgregoribattista** [comment]
not merge yet

**Pull request merged**